src/inference/include/openvino/runtime/properties.hpp

```diff
@@ -0,0 +1,638 @@
+
+namespace log {
+
+/**
+ * @brief Enum to define possible affinity template hints
+ */
+enum class Level {
+    NO = -1,      //!< disable any loging
+    ERR = 0,      //!< error events that might still allow the application to continue running
+    WARNING = 1,  //!< potentially harmful situations which may further lead to ERROR
+    INFO = 2,     //!< informational messages that display the progress of the application at coarse-grained level
+    DEBUG = 3,    //!< fine-grained events that are most useful to debug an application.
+    TRACE = 4,    //!< finer-grained informational events than the DEBUG
+};
+
+/** @cond INTERNAL */
+inline std::ostream& operator<<(std::ostream& os, const Level& level) {
+    switch (level) {
+    case Level::NO:
+        return os << "NO";
+    case Level::ERR:
+        return os << "LOG_ERROR";
+    case Level::WARNING:
+        return os << "LOG_WARNING";
+    case Level::INFO:
+        return os << "LOG_INFO";
+    case Level::DEBUG:
+        return os << "LOG_DEBUG";
+    case Level::TRACE:
+        return os << "LOG_TRACE";
+    default:
+        throw ov::Exception{"Unsupported log level"};
+    }
+}
+
+inline std::istream& operator>>(std::istream& is, Level& level) {
+    std::string str;
+    is >> str;
+    if (str == "NO") {
+        level = Level::NO;
+    } else if (str == "LOG_ERROR") {
+        level = Level::ERR;
+    } else if (str == "LOG_WARNING") {
+        level = Level::WARNING;
+    } else if (str == "LOG_INFO") {
+        level = Level::INFO;
+    } else if (str == "LOG_DEBUG") {
+        level = Level::DEBUG;
+    } else if (str == "LOG_TRACE") {
+        level = Level::TRACE;
+    } else {
+        throw ov::Exception{"Unsupported log level: " + str};
+    }
+    return is;
+}
+/** @endcond */
+
+/**
+ * @brief the key for setting desirable log level.
+ */
+static constexpr Key<Level> level{"LOG_LEVEL"};
+}  // namespace log
+
+/**
+ * @brief This key defines the directory which will be used to store any data cached by plugins.
+ *
+ * The underlying cache structure is not defined and might differ between OpenVINO releases
+ * Cached data might be platform / device specific and might be invalid after OpenVINO version change
+ * If this key is not specified or value is empty string, then caching is disabled.
+ * The key might enable caching for the plugin using the following code:
+ *
+ * @code
+ * ie.set_property("GPU", ov::cache_dir("cache/")); // enables cache for GPU plugin
+ * @endcode
+ *
+ * The following code enables caching of compiled network blobs for devices where import/export is supported
+ *
+ * @code
+ * ie.set_property(ov::cache_dir("cache/")); // enables models cache
+ * @endcode
+ */
+static constexpr Key<std::string> cache_dir{"CACHE_DIR"};
+
+namespace device {
+
+/**
+ * @brief the key for setting of required device to execute on
+ * values: device id starts from "0" - first device, "1" - second device, etc
+ */
+static constexpr Key<std::string> id{"DEVICE_ID"};
```

Comment:
1. it can be passed to compile_model, so is it valid to call it default?
2. Good note

Reply:
> it can be passed to compile_model, so is it valid to call it default?

Good question. I think it would be logical if all 3 calls below do the same thing (loading to GPU.2)
1. compile_model(model, "GPU", device::default_id("2")) 
2. compile_model(model, "GPU.2"); 
3. compile_model(model, "GPU.2", device::default_id("1") );

But I believe that currently the third call will actually load to GPU.1. In that case it would be weird to call it "default_id'. So if we're not going to change the behavior w.r.t. this option, then we can keep the `id` name.